.github/workflows/default_on_push.yml

```diff
@@ -77,6 +83,12 @@ jobs:
         with:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
```

Comment:
I've been doing some similar tests as well (unsuccessfully) - in your testing, did this work end-to-end and result in the image going into GHCR? When I attempt this, it authenticates successfully at login, but on push to registry it gets a 403. I've given the PAT in my tests the read/write:packages scope per docs. Curious if you've had this work in practice yet.

Additionally, the username line will need to change to a secret for the user whose PAT we use for this. This variable as-is works fine for user-owned repos, but for org-repos, you'll be authenticating with someone's specific PAT, and you need that username in the login.

Reply:
Yes, running into the same issues first but was able to solve it by enabling the improved container support for my personal account. This would be needed for the organization as well ([docs](https://docs.github.com/en/packages/guides/enabling-improved-container-support)). I will add it to the initial description as it provides the whole overview.

Here is the output of my private test:
```
#6 exporting to image
#6 exporting layers
#6 exporting layers 0.2s done
#6 exporting manifest sha256:42daa0610f0df6ad71fa4288d0335083e7958116fdc31a8f51102272cb3113f9 done
#6 exporting config sha256:ae1f6955dbdff627e8eebebccc8704f08d90bb3364d24f083f8e4613c54d5166 done
#6 pushing layers
#6 pushing manifest for docker.io/wernerfred/testing:edge
#6 pushing manifest for docker.io/wernerfred/testing:edge 0.1s done
#6 pushing layers 1.5s done
#6 pushing manifest for ghcr.io/wernerfred/testing:edge
#6 pushing manifest for ghcr.io/wernerfred/testing:edge 0.5s done
#6 DONE 3.0s
```

> Additionally, the username line will need to change to a secret for the user whose PAT we use for this. This variable as-is works fine for user-owned repos, but for org-repos, you'll be authenticating with someone's specific PAT, and you need that username in the login.

Thanks for that hint I will add `${{ secrets.GHCR_USERNAME }}`

Reply:
Solved with https://github.com/docker-mailserver/docker-mailserver/pull/1763/commits/cb8034262a981d3a197fa467dcdc7fe5c4327262